servers/src/main/java/tachyon/master/MasterBase.java

```diff
@@ -0,0 +1,166 @@
+
+  private final Journal mJournal;
+  private final ExecutorService mExecutorService;
+
+  /** true if this master is in leader mode, and not standby mode. */
+  private boolean mIsLeader = false;
+
+  /** The thread that tails the journal when the master is in standby mode. */
+  private JournalTailerThread mStandbyJournalTailer = null;
+
+  private JournalWriter mJournalWriter = null;
+
+  protected MasterBase(Journal journal, ExecutorService executorService) {
+    mJournal = Preconditions.checkNotNull(journal);
+    mExecutorService = Preconditions.checkNotNull(executorService);
+  }
+
+  @Override
+  public void processJournalCheckpoint(JournalInputStream inputStream) throws IOException {
+    JournalEntry entry;
+    while ((entry = inputStream.getNextEntry()) != null) {
+      processJournalEntry(entry);
+    }
+    inputStream.close();
+  }
+
+  @Override
+  public void start(boolean isLeader) throws IOException {
+    LOG.info("Starting master. isLeader: " + isLeader);
+    mIsLeader = isLeader;
+    if (mIsLeader) {
+      // Replay all the state of the checkpoint and the completed log files.
+      // TODO: only do this if this is a fresh start, not if this master had already been tailing
+      // the journal.
+      LOG.info(getProcessorName() + ": process completed logs before becoming master.");
+      JournalTailer catchupTailer = new JournalTailer(this, mJournal);
+      boolean checkpointExists = true;
+      try {
+        catchupTailer.getCheckpointLastModifiedTimeMs();
+      } catch (IOException ioe) {
+        // The checkpoint doesn't exist yet. This is probably the first execution ever, or this is a
+        // testing master.
+        checkpointExists = false;
+      }
+      if (checkpointExists) {
+        catchupTailer.processJournalCheckpoint(true);
+        catchupTailer.processNextJournalLogFiles();
+      }
+
+      // initialize the journal and write out the checkpoint file (the state of all completed logs).
+      mJournalWriter = mJournal.getNewWriter();
+      writeToJournal(mJournalWriter.getCheckpointOutputStream());
+      mJournalWriter.getCheckpointOutputStream().close();
+
+      // Final catchup stage. The last in-progress file (if it existed) was marked as complete when
+      // the checkpoint file was closed. That last completed file must be processed to get to the
+      // latest state. Read and process the completed file.
+      LOG.info(getProcessorName() + ": process the last completed log before becoming master.");
+      catchupTailer = new JournalTailer(this, mJournal);
+      catchupTailer.processJournalCheckpoint(false);
+      catchupTailer.processNextJournalLogFiles();
+    } else {
+      // in standby mode. Start the journal tailer thread.
+      mStandbyJournalTailer = new JournalTailerThread(this, mJournal);
+      mStandbyJournalTailer.start();
+    }
+  }
+
+  @Override
+  public void stop() throws IOException {
+    LOG.info("Stopping master. isLeader: " + isLeaderMode());
```

Comment:
What's the convention on using the getter methods internally in the method? It seems less straightforward.